### DISCUSSION TITLE
Special use case NFT transfer with off chain payment

### DISCUSSION THREAD

**Discussion by rarepepi** [ProjectOpenSea/seaport#1288] [Q&A]
**GM degens!**

I have a very particular and I think interesting use case for **Credit Card based purchases of NFTs on Seaport**. I would really love to get everyone's thoughts on what would be the most efficient and secure solution here.

**Idea:**
1. Seller signs `approvalForAll` to the marketplace contract for the NFT they want to sell 
2. Buyer comes to the marketplaces, chooses credit card checkout and pays using fiat (i have the p2p fiat rails already for getting the seller their money)
3. The marketplace contract calls an `adminTransfer` function that transfers the NFT from the seller to the buyer

**Results:**
1. Seller was able to get paid directly in fiat into their bank account
2. Buyer was able to purchase the NFT without having to onboard into Web3(download coinbase, get eths, transfer to a metamask, etc)

It would be so amazing if the community could help me figure out the best way around this, while being as secure as possible.

Any feedback would be immensely appreciated!

Thanks!

**Comment by ryanio** [comment]
Having an adminTransfer function is not secure or trustless. The best way to achieve this is how e.g. Moonpay works on OpenSea works today, which is that you pay Moonpay via credit card and they'll complete the sale on chain by fulfilling the listing and transferring the NFT to you.

**Reply by rarepepi** [reply]
Firstly, thanks for answering Ryan I was listening to all your talks on Seaport today, and was thinking of how I could contact you haha.

Completely understand, however, flow of funds such as Moonpay's requires multiple txns (extra gas charged to end user/buyer), large spreads when converting from fiat to the native token, and lengthy KYC involvement since fungibles are considered money. What I am proposing is a new way of doing fiat onboarding, which is not as cumbersome. One in which the seller is accepting fiat in a direct p2p manner off-chain and simply transferring the ownership of the NFT on-chain.

On the trustless part, I completely agree which is exactly why I'm posing this question. I have thought about methods such as committing valid off-chain payment IDs on-chain and requiring `adminTransfer` to verify a Merkle proof against it, or maybe using something like Chainlink to verify the valid off-chain payment through an API call from within the contract. Would really love to get your thoughts about the best combination of this new flow of funds which is 10x better for onboarding brand new users into crypto, while also maintaining trustlessness within the marketplace contract. Thank you I really appreciate it!

**Reply by rarepepi** [reply]
@ryanio If I did want to add a transfer function that requires a 712 signature that I can create, where would I put in Seaport? Would it be a conduit class overwrite or zones maybe?
